modin/core/dataframe/base/partitioning/axis_partition.py

```diff
@@ -14,9 +14,10 @@
 """Base class of an axis partition for a Modin Dataframe."""
 
 from abc import ABC
+from modin.logging import ClassLogger
 
 
-class BaseDataframeAxisPartition(ABC):  # pragma: no cover
+class BaseDataframeAxisPartition(ClassLogger, ABC):  # pragma: no cover
```

Comment:
I wonder if we should set some specific `modin_layer="PARTITION"` logging layer here (as `ClassLogger` supports this), to make following logs easier

Here and in other classes, too. Stuffing everything under `PANDAS-API` umbrella feels wrong.

Reply:
+1 to @vnlitvinov I think we want to make sure that log column has some meaning 😄

Reply:
I tried doing `modin_layer="PARTITION"` in the class declaration as you mentioned, but the `modin_layer` variable was not being propagated to all subclasses. E.g. `PandasDataframePartition` would be `PARTITION` but `PandasOnRayDataframePartition` would be `PANDAS-API`.

I think the reason is that `__init_subclass__` is called from the perspective of `cls` (`PandasOnRayDataframePartition`) to `ClassLogger`, so it doesn't have the `modin_layer` of its immediate parent, `PandasDataframePartition`. Perhaps we could add this to the class declaration for each subclass. What I've implemented is another way, which is to "propagate" the var by making it a class variable, and making `ClassLogger` retrieve that class variable.